src/Interpreters/Cache/FileSegment.cpp

```diff
@@ -458,26 +463,24 @@ bool FileSegment::reserve(size_t size_to_reserve)
     bool reserved = already_reserved_size >= size_to_reserve;
     if (!reserved)
     {
-        std::lock_guard cache_lock(cache->mutex);
-        std::lock_guard segment_lock(mutex);
-
         size_to_reserve = size_to_reserve - already_reserved_size;
 
         if (is_unbound && is_file_segment_size_exceeded)
-        {
             segment_range.right = range().left + expected_downloaded_size + size_to_reserve;
```

Comment:
this method is not used here, we change class member directly here. Nevertheless, there is no point in returning a copy even if this method was used here, because then we will change the copy, which does not make sense.

Reply:
Ok, indeed better to add a mutex, but in fact no race is possible here, because range can be changed only in case of feature "temporary files in cache" (tmp files for joins) and in this feature cache is used single threaded way. In all other cases range is immutable.